index.js

```diff
@@ -2120,24 +2124,41 @@
   //. Cons('red', Cons('blue', Cons('green', Nil)))
   //. ```
   function sortBy(f, foldable) {
-    var rs = reduce(function(xs, x) {
-      var fx = f(x);
-      var lower = 0;
-      var upper = xs.length;
-      while (lower < upper) {
-        var idx = Math.floor((lower + upper) / 2);
-        if (lte(xs[idx].fx, fx)) lower = idx + 1; else upper = idx;
-      }
-      xs.splice(lower, 0, {x: x, fx: fx});
+    var foldableIsArray = Array.isArray(foldable);
+    var idx = 0;
+    function reducer(xs, x) {
+      xs.push({idx: idx, x: x, prop: f(x)});
+      idx += 1;
       return xs;
-    }, [], foldable);
-
-    var F = foldable.constructor;
-    var result = empty(F);
-    for (var idx = 0; idx < rs.length; idx += 1) {
-      result = concat(result, of(F, rs[idx].x));
     }
-    return result;
+    var rs = reduce(reducer, [], foldable);
+    if (rs.length === 0) {
+      return foldable;
+    }
```

Comment:
Does this special case make an appreciable difference? I would have thought that sorting an empty structure would be an efficient operation.

Reply:
I now see that this is guarding `rs[0].prop` below. I would still like to remove the special case. We could define `lteFn` like so:

```javascript
var type = rs.length > 0 ? typeof rs[0].prop : 'undefined';
var lteFn = type === 'number' || type === 'string' ? ... : lte;
```

Reply:
It's because he's unsafely accessing head in other path.

Reply:
:zap: